src/main/scala/parsing/Parser.scala

```diff
@@ -36,20 +37,19 @@ object Parser extends Encoding {
           parsedLines = droppedLines.map(_.merge),
           droppedLines = parsedLines.map(_.merge)
         )
-      case ReaderType(input)   =>
-        val source: Source =  Source.fromString(input.toString)
-        val sourceLines: Iterator[String] = source.getLines()
-        val parsed: Seq[Either[EncodeType, EncodeType]] = sourceLines.toList.map(parsingLogic)
-        source.close()
+      case ReaderType(input) =>
+        val iterator = Iterator.continually(input.read().toChar).toList
+        val lines = iterator.map(_.toString)
```

Comment:
`_.toString` here will make the whole file to be in a single String, it does not separate into each line into a seperate String

Reply:
I recommend:
```suggestion
        val chars = Iterator.continually(input.read().toChar).toArray
        val lines = Source.fromChars(chars)
```
an then replicate Source implementation

Reply:
thank you for your suggestion joão , il try to do it